packages/pigeon/lib/objc_generator.dart

```diff
@@ -384,11 +425,22 @@ String _dictValue(
   }
 }
 
+String _getSelector(Method func, String lastSelectorComponent) {
+  final Iterable<String> selectorComponents = lastSelectorComponent.isEmpty
+      ? _getSelectorComponents(func)
+      : followedByOne(_getSelectorComponents(func), lastSelectorComponent);
+  return selectorComponents.join(':') + ':';
+}
+
+/// Returns an argument name that can be used in a context where it is possible to collide.
+String _getSafeArgName(int count, NamedType arg) =>
+    arg.name.isEmpty ? 'arg$count' : 'arg_${arg.name}';
+
 void _writeHostApiSource(Indent indent, ObjcOptions options, Api api) {
   assert(api.location == ApiLocation.host);
   final String apiName = _className(options.prefix, api.name);
   indent.write(
-      'void ${apiName}Setup(id<FlutterBinaryMessenger> binaryMessenger, id<$apiName> api) ');
+      'void ${apiName}Setup(id<FlutterBinaryMessenger> binaryMessenger, NSObject<$apiName>* api) ');
```

Comment:
Nit: C function names should generally be verbs, and the verb is "set up" (and thus `SetUp(...)`)

Reply:
Yea, good point.  It is consistent across platforms and unfortunately the `set` verb has special connotations in some languages too.  Like we aren't setting `up`.

Reply:
> It is consistent across platforms

The comment is actually applicable across platforms/languages as well, I just flagged it in the one place. (Java style guide: "Method names are typically verbs or verb phrases." Effective Dart: "PREFER starting function or method comments with third-person verbs.")

> and unfortunately the set verb has special connotations in some languages too. Like we aren't setting up.

There are a few options to avoid that:
- Stick with setup as a noun, but add a verb: `PerformSetup`/`DoSetup`/etc.
- Make the name less generic: `SetUpPigeonMethodChannels` or the like.

In this case I missed the `${apiName}` being the start, so it should be the first. For Dart it could be either.